dist/tools/kconfiglib/examples/select_specific_implementation_from_module/README.md

```diff
@@ -0,0 +1,74 @@
+# Select a specific choice implementation from a module
+
+This is a design pattern when one must select a specific implementation from
+a module. This is specifically for modules that are not defined in boards or
+applications.
+
+## Reasoning
+
+We need a choice to ensure mutual exclusive options. The issue is that we need
+to detect if multiple modules `REQUIRES_*` different choices. If that happens
+then it will select the first fitting choice in the list but we can use the
+`ERROR_*` convention to enforce the correct one is being used. The example also
+allows for auto-selection based on the `HAS_*` convention.
+
+### Use case
+
+The use case is for something like having `MODULE_STDIO` selecting a specific
+`USB` implementation because it `REQUIRES_CDC_ACM` feature that may not exist
+in all `USB` implementations.
+
+## Usage
+
+To view how it operates we can directly invoke the `menuconfig` from this
+directory.
+
+This will show the basic operation without anything selected.
+
+
+```
+./../../riot_menuconfig.py
+```
+
+
+### Pre-selected automatic selection based on `HAS_*`
+
+```
+KCONFIG_CONFIG=app.auto.config ./../../riot_menuconfig.py
+```
+
+### Pre-selected the `HAS_*` being overridden by the `MODULE_*` requirements
+
+```
+KCONFIG_CONFIG=app.has_override.config ./../../riot_menuconfig.py
+```
+
+### Pre-selected an error condition having a module select impossible requirements
+
+```
+KCONFIG_CONFIG=error.config ./../../riot_menuconfig.py
+```
+
+## Other tooling
+
+The `Kconfig.generated` shows the actual design pattern and is generated with
+the `generate_example.py` script.
+
+For more details, try:
+```
+generate_example.py -h
+```
+
+## FAQs
+
+### Why can't I just select the choice option?
+
+This is a "feature" of Kconfig, problems would then occur when multiple choice
+options are selected since a choice can only select one.  It is possible in the
+`app.config` settings though...
```

Comment:
This sounds a bit misleading...`choice` options will *always* be mutually exclusive. What is possible from `app.config` files (and menuconfig or other interfaces) is to -attempt to- set one of the options to `y`